common/models/templates/test/test-trie-compression.js

```diff
@@ -0,0 +1,260 @@
+      key: 'four',
+      content: 'four',
+      weight: 8
+    }
+  }
+};
+
+TEST_DATA.LEAVES = {
+  four: {
+    // expected width difference: 5 (2: total size, 2: weight, 1: entry count)
+    //                   -totalLen-              -weight-              -type=leaf + size-
+    compressed: `${compressNumber(13, 2)}${compressNumber(8, 2)}${compressNumber(0x8000 + 1, 1)}${TEST_DATA.ENTRIES.four.compressed}`,
+    decompressed: {
+      type: 'leaf',
+      weight: 8,
+      entries: [TEST_DATA.ENTRIES.four.decompressed]
+    },
+    original: {
+      type: 'leaf',
+      weight: 8,
+      entries: [{
+        key: 'four',
+        content: 'four',
+        weight: 8
+      }]
+    }
+  }
+}
+
+TEST_DATA.NODES = {
+  four: {
+    // expected width difference: 6 (2: total size, 2: weight, 1: entry count, 1: value count)
+    //             -totalLen-  -weight- -type/size-
+    compressed: `${compressNumber(19, 2)}${compressNumber(8, 2)}${compressNumber(1)}r${TEST_DATA.LEAVES.four.compressed}`,
+    decompressed: {
+      type: 'internal',
+      weight: 8,
+      values: ['r'],
+      children: {r: TEST_DATA.LEAVES.four.compressed}
+    },
+    original: {
+      type: 'internal',
+      weight: 8,
+      values: ['r'],
+      children: {r: TEST_DATA.LEAVES.four.decompressed}
+    }
+  }
+}
+
+describe('Trie compression', function() {
+  describe('`number`s', () => {
+    it('uses single-char compression by default', () => {
+      assert.equal(compressNumber(0x0020).length, 1);
+    });
+
+    it('compresses properly when targeting single-char width', () => {
+      assert.equal(compressNumber(0x0020, 1), String.fromCharCode(0x0020 + ENCODED_NUM_BASE));
+      assert.equal(compressNumber('"'.charCodeAt(0), 1), String.fromCharCode('"'.charCodeAt(0) + ENCODED_NUM_BASE));
+    });
+
+    it('has non-null leading char for numbers needing two-char representations', () => {
+      assert.notEqual(compressNumber(0x00200020, 2).charAt(0), String.fromCharCode(0));
+      assert.notEqual(
+        compressNumber(0x0321ad20, 2).charAt(0), String.fromCharCode(0)
+      );
+    });
+
+    it('width 2: compressing values one-char wide', () => {
+      assert.equal(compressNumber(0x0020, 2), `${String.fromCharCode(ENCODED_NUM_BASE)}${String.fromCharCode(0x0020 + ENCODED_NUM_BASE)}`);
+    });
+
+    it('throws when numbers are too large for the specified width', () => {
+      assert.throws(() => compressNumber(0x00200020, 1));
+      assert.throws(() => compressNumber(0x002000200020, 2));
+    })
+  });
+
+  describe('`Entry`s', () => {
+    it('compresses properly', () => {
+      assert.equal(compressEntry(TEST_DATA.ENTRIES.four.original), TEST_DATA.ENTRIES.four.compressed);
+    });
+  });
+
+  describe('Leaf nodes', () => {
+    it('compresses (mocked Entry)', () => {
+      // Should not attempt to recompress the mock-compressed entry.
+      assert.equal(compressNode(TEST_DATA.LEAVES.four.decompressed), TEST_DATA.LEAVES.four.compressed);
```

Comment:
I do not see a difference between `TEST_DATA.LEAVES.four.decompressed` and `TEST_DATA.LEAVES.four.original`, so the comment doesn't make sense to me and I don't see the difference between the two tests here.